src/pages/countdown.js

```diff
@@ -0,0 +1,84 @@
+import {useMemo} from 'react';
+
+import DumbbellIcon from '../components/Icons/DumbbellIcon';
+import StyledDumbbellIcon from '../components/Icons/StyledDumbbellIcon';
+import MoveBackButton from '../components/MoveBackButton';
+import {StyledHeader, StyledH1} from '../components/StyledHeader';
+import StyledLayout from '../components/StyledLayout';
+import TimerCard from '../components/TimerCard';
+import useStore from '../hooks/useStore';
+
+export default function TimerContainer() {
+	const currentExercise = useStore(state => state.currentExercise);
+
+	const timers = useMemo(() => {
```

Comment:
Why do you use `useMemo` here? I think you do not need it 🤔

Reply:
thank you for checking! I thought it would be nice to use it since i only want to render the times and amounts once and not every single time new if there are other things happening on the countdown page, in the future if i work with a save workout button work for example it should prevent extra rendering or unnecessary loading :) i am not 100% sure but i thought it would make sense, i could also use useEffect but than it renders everytime some inteaction is happening.

Reply:
I think this is no use-case for `useMemo`.

The hook can be used, when you have a component with **multiple** states and a function, that should be called, when a **specific state** changes (but not always, when any state changes).

With `useMemo` you can tie a function to a specific state, so that the function is re-called only when this state changes. The same concepts can be used with a functions that uses multiple states.

In your case there is only one state `currentExercise` and your `timer` function depends on this state. Therefore `useMemo` doesn't provide any benefit.

You can rad about `useMemo` here:
https://reactjs.org/docs/hooks-reference.html#usememo
https://usehooks.com/useMemo/